apps/dashr-oil-and-gas/app.R

```diff
@@ -0,0 +1,829 @@
+              list(dccGraph(id="count_graph")),
+              id="countGraphContainer",
+              className="pretty_container"
+            )
+          ),
+          id="right-column",
+          className="eight columns"
+        )
+      ),
+      className="row flex-display"
+    ),
+    htmlDiv(
+      list(
+        htmlDiv(
+          list(dccGraph(id="main_graph")),
+          className="pretty_container seven columns"
+        ),
+        htmlDiv(
+          list(dccGraph(id="individual_graph")),
+          className="pretty_container five columns"
+        )
+      ),
+      className="row flex-display"
+    ),
+    htmlDiv(
+      list(
+        htmlDiv(
+          list(dccGraph(id="pie_graph")),
+          className="pretty_container seven columns"
+        ),
+        htmlDiv(
+          list(dccGraph(id="aggregate_graph")),
+          className="pretty_container five columns"
+        )
+      ),
+      className="row flex-display"
+    )
+  ),
+  id="mainContainer",
+  style=list("display"= "flex", "flex-direction"= "column")
+))
+
+filter_Dataframe <- function(df, 
+                             well_statuses, 
+                             well_types, 
+                             year_slider){
+  df <- as.data.table(df)
+  min_year <- as.Date(paste0(year_slider[[1]], "-01-01"))
+  max_year <- as.Date(paste0(year_slider[[2]], "-01-01"))
+  df <- df[df$Well_Status %in% well_statuses & df$Well_Type %in% well_types,]
+  df <- df[df$Date_Well_Completed > min_year & df$Date_Well_Completed < max_year, ]
+  return(df)
+}
+
+fetch_individual <- function(api){
+  
+  index = as.list(c(min(dfIn[dfIn$API_WellNo == api,'Reporting.Year']):
+                      max(dfIn[dfIn$API_WellNo == api,'Reporting.Year'])))
+  gas = list()
+  oil = list()
+  water = list()
+  
+  for (year in index) {
+    
+    gas[[year]]  <- dfIn[dfIn$API_WellNo == api & dfIn$Reporting.Year == year,'Gas.Produced..MCF']
+    oil[[year]]  <- dfIn[dfIn$API_WellNo == api & dfIn$Reporting.Year == year,'Oil.Produced..bbl']
+    water[[year]] <- dfIn[dfIn$API_WellNo == api & dfIn$Reporting.Year == year,'Water.Produced..bbl']
+  }
+  return(list(index,water[c(unlist(index))], oil[c(unlist(index))], gas[c(unlist(index))]))
+}
+
+fetch_aggregate <- function(dff,year_slider){
+  gas1 = 0
+  oil1 = 0
+  water1 = 0
+  min_year <- paste0(year_slider[[1]], "-01-01")
+  
+  index = max(year(as.Date(min_year)):1985):2017
+  
+  gas = list()
+  oil = list()
+  water = list()  
+  inn <- merge(dff,dfIn, by = 'API_WellNo')
+  for (year in index) {
+    count_gas = 0
+    count_oil = 0
+    count_water = 0 
+    count_gas <- sum(inn$Gas.Produced..MCF[inn$Reporting.Year %in% year])
+    count_water <- sum(inn$Water.Produced..bbl[inn$Reporting.Year %in% year])
+    count_oil <- sum(inn$Oil.Produced..bbl[inn$Reporting.Year %in% year])
+    gas[[year]] = count_gas
+    water[[year]] = count_water
+    oil[[year]] = count_oil
+  }
+  gas1 = sum(unlist(gas[c(index)]))
```

Comment:
I think you could potentially write this as

`gas1 <- Reduce("+", gas[index])`